crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -49,17 +49,19 @@ const GAS_PRICE_BUMP: f64 = 1.125;
 
 /// Parameters for transaction submitting
 #[derive(Clone, Default)]
-pub struct SubmitterParams {
+pub struct SubmitterParams<'a> {
     /// Desired duration to include the transaction in a block
     pub target_confirm_time: Duration, //todo ds change to blocks in the following PR
     /// Estimated gas consumption of a transaction
     pub gas_estimate: U256,
     /// Maximum duration of a single run loop
     pub deadline: Option<Instant>,
-    /// Resimulate and resend transaction on every retry_interval seconds
+    /// Re-simulate and resend transaction on every retry_interval seconds
     pub retry_interval: Duration,
     /// Network id (mainnet, rinkeby, goerli, gnosis chain)
     pub network_id: String,
+    /// Additional bytes to append to the call data. This is required by the `driver`.
+    pub additional_call_data: &'a [u8],
```

Comment:
> We can't use tx hash instead of solution ID because you need some id before entering the solving part right?

Also, these are _in theory_ not unique.

Imagine:
- Solver A wins auction X
- Solver A submits settlement X to Flashbots
- The transaction doesn't get mined in time
- Solver A wins auction X+1 with the exact same solution (so same tx hash)
- Solver A submits settlement X+1 to Flashbots
- Transaction gets mined - no way of knowing if it was from auction X or X+1